Explorer/Assets/DCL/SDKComponents/TextShape/System/InstantiateTextShapeSystem.cs

```diff
@@ -0,0 +1,55 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.Throttling;
+using DCL.ECSComponents;
+using DCL.Optimization.PerformanceBudgeting;
+using DCL.Profiling;
+using DCL.SDKComponents.TextShape.Component;
+using DCL.SDKComponents.TextShape.Renderer.Factory;
+using ECS.Abstract;
+using ECS.Unity.Groups;
+using ECS.Unity.Transforms.Components;
+
+namespace DCL.SDKComponents.TextShape.System
+{
+    [UpdateInGroup(typeof(ComponentInstantiationGroup))]
+    [ThrottlingEnabled]
+    public partial class InstantiateTextShapeSystem : BaseUnityLoopSystem
+    {
+        private readonly ITextShapeRendererFactory textShapeRendererFactory;
+        private readonly IConcurrentBudgetProvider instantiationFrameTimeBudgetProvider;
+
+        public InstantiateTextShapeSystem(World world, ITextShapeRendererFactory textShapeRendererFactory, IConcurrentBudgetProvider instantiationFrameTimeBudgetProvider) : base(world)
+        {
+            this.textShapeRendererFactory = textShapeRendererFactory;
+            this.instantiationFrameTimeBudgetProvider = instantiationFrameTimeBudgetProvider;
+        }
+
+        public InstantiateTextShapeSystem(World world, ITextShapeRendererFactory textShapeRendererFactory) : this(
+            world,
+            textShapeRendererFactory,
+            new FrameTimeCapBudgetProvider(
```

Comment:
Isn't 33ms most of the timeframe? Should this be lower?

Reply:
I took this value from DCL.PluginSystem.Global.StaticSettings as default

```
// Performance budgeting
[field: Header("Performance Budgeting")] [field: Space]
[field: SerializeField]
public int FrameTimeCap { get; private set; } = 33; // in [ms]. Table: 33ms ~ 30fps | 16ms ~ 60fps | 11ms ~ 90 fps | 8ms ~ 120fps
```